modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-sets/FieldSetModal.js

```diff
@@ -0,0 +1,472 @@
+					...fieldSetProp,
+					availableLanguageIds,
+					dataDefinitionFields,
+					defaultDataLayout: normalizeDataLayout(
+						{
+							...fieldSetProp.defaultDataLayout,
+
+							dataLayoutPages: dataLayoutPages.map((page) => {
+								return {
+									...page,
+									description: {
+										[defaultLanguageId]: '',
+									},
+
+									title: {
+										[defaultLanguageId]: '',
+									},
+								};
+							}),
+
+							name,
+						},
+						defaultLanguageId
+					),
+					name,
+				};
+
+				const onPropagate = async () => {
+					const savedFieldset = await updateItem(
+						`/o/data-engine/v2.0/data-definitions/${dataDefinitionId}`,
+						updatedFieldSet
+					);
+
+					updateFieldSetList({fieldSet: savedFieldset});
+					successToast(Liferay.Language.get('fieldset-saved'));
+				};
+
+				onClose();
+				onUpdate({onPropagate, updatedFieldSet});
+			}
+			else {
+
+				// create new fieldSet from scratch
+
+				const fieldSet = {
+					availableLanguageIds,
+					dataDefinitionFields,
+					defaultDataLayout: {
+						dataLayoutPages,
+						name,
+					},
+					name,
+				};
+
+				const updatedFieldSet = await addItem(
+					`/o/data-engine/v2.0/data-definitions/by-content-type/${contentType}`,
+					fieldSet
+				);
+				onClose();
+				updateFieldSetList({fieldSet: updatedFieldSet, isAdding: true});
+				successToast(Liferay.Language.get('fieldset-saved'));
+			}
+		}
+		catch ({message}) {
+			errorToast(message);
+		}
+	};
+
+	const isFieldSetInvalid = () =>
+		Object.keys(name).length == 0 ||
+		!dataLayout?.dataLayoutPages ||
+		isDataLayoutEmpty(dataLayout.dataLayoutPages);
+
+	return (
+		<>
+			<ClayModal.Header>
+				{dataDefinitionId
+					? Liferay.Language.get('edit-fieldset')
+					: Liferay.Language.get('create-new-fieldset')}
+			</ClayModal.Header>
+			<ClayModal.Header withTitle={false}>
+				<TranslationManager
+					defaultLanguageId={defaultLanguageId}
+					editingLanguageId={editingLanguageId}
+					onEditingLanguageIdChange={changeEditingLanguageId}
+					translatedLanguageIds={name}
+				/>
+				<ClayInput.Group className="pl-4 pr-4">
+					<ClayInput.GroupItem>
+						<ClayInput
+							aria-label={Liferay.Language.get(
+								'untitled-fieldset'
+							)}
+							autoFocus
+							className="form-control-inline"
+							dir={
+								Liferay.Language.direction[editingLanguageId] ||
+								'ltr'
+							}
+							onChange={({target: {value}}) =>
+								setName((name) => ({
+									...name,
+									[editingLanguageId]: value,
+								}))
+							}
+							placeholder={Liferay.Language.get(
+								'untitled-fieldset'
+							)}
+							type="text"
+							value={
+								name[editingLanguageId] ??
+								name[defaultLanguageId] ??
+								''
+							}
+						/>
+					</ClayInput.GroupItem>
+				</ClayInput.Group>
+			</ClayModal.Header>
+			<ClayModal.Body>
+				<div className="pl-4 pr-4">
+					<ClayModalProvider>
+						<DndProvider backend={HTML5Backend}>
```

Comment:
Could it be that we are no longer under the custody of those providers that are already declared in the `App`? If it runs on the same tree as React and is below the providers of the `App` we do not need to declare it again in this case because we can share.

- `DndProvider` it will have no effect here, under the hood he uses the globally declared Provider, he is treated as a singleton